pages/[id].jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import utilStyles from '../styles/categorypage.module.css'
+import Layout from "../components/Layout"
+import ModalStaet from '../components/ModalStaet'
+import Image from 'next/image'
+import { useDispatch,useSelector } from "react-redux";
+import { deleteCategory } from "../provider/dishesSlice";
+import { useSelect } from "../hooks/useSelectState";
+
+
+export async function getStaticPaths() {
+  return {
+    paths:[
+      { params: { id: 'all'} },
+      { params: { id: 'meat'} },
+      { params: { id: 'fish'} },
+      { params: { id: 'noodle'} },
+      { params: { id: 'salad'} },
+      { params: { id: 'dessert'} },
+      { params: { id: 'coffee'} }
+    ],
+    fallback: false
+  }
+}
+
+export const getStaticProps = async context => {
+  const { id } = context.params 
+  const categoryName = `${id}Category`
+  
+  return {
+    props: { id, categoryName }
+  }
+}
+
+const Categorypage = ({ id, categoryName }) => {
+
+  const [modal, setModal] = useState(false);
+  const { onSelectState, selectedState } = useSelect();
+
+  const onClickModal = () => {
+    setModal(!modal);
+  }
+
+  const onClickOpen = (i) => {
+    onSelectState({ allCategory, i });
+    onClickModal();
+  }
+
+  const dispatch = useDispatch();
+
+  const { 
+    allCategory,
+    meatCategory,
+    fishCategory,
+    noodleCategory,
+    saladCategory,
+    dessertCategory,
+    coffeeCategory
+  } = useSelector((state) => state.dishes);
+
+  const onClickDelete = (i) => {
+    dispatch(deleteCategory(i));
+  }
+
+  const categoryMap = {
+    allCategory: allCategory,
+    meatCategory: meatCategory,
+    fishCategory: fishCategory,
+    noodleCategory: noodleCategory,
+    saladCategory: saladCategory,
+    dessertCategory: dessertCategory,
+    coffeeCategory: coffeeCategory
+  };
```

Comment:
以下のようにすればgetStaticPropsでわざわざ `~Category` とつけなくて良くなるので良いのでは？
そもそもcategoryMapのプロパティである時点でカテゴリーで有ることは明白なので、変に `~category` にしなくていいです。
名前つけるときは無駄に冗長でないかどうかを考えてからつけましょう。

```suggestion
  const categoryMap = {
    all: allCategory,
    meat: meatCategory,
    fish: fishCategory,
    noodle: noodleCategory,
    salad: saladCategory,
    dessert: dessertCategory,
    coffee: coffeeCategory
  };
```

Reply:
また、これはどっちでも良いのですが、idをstringからenumにすることで、Switch文でいい感じにcategoryを抽出できます。
この方法だとTypeScriptの型システムの恩恵で、将来IDを追加した際にSwitch文の方も改修しないとエラーが出るようになるので事前にバグを防げます。
現在の方法だとID追加時に忘れずにcategoryMapの改修をしないと行けないので、運用上で注意が必要ですよね。

Reply:
idをstringからenumにすることで、Switch文でいい感じにcategoryを抽出すると言うのは、getStaticProps の中で
Switchでcategoryを抽出するのでしょうか。